### ISSUE TITLE
is there a way to specify password for vncviewer from command line on windows?

### ISSUE THREAD

**Issue by rpodgorny** [TigerVNC/tigervnc#883]
you can specify a path to passwd file but that has to be created using vncpasswd which (afaik) only has build for unix.

also, being able to omit vncpasswd completely would be just great (for automation)! read the password from env...? from stdin...? or just specify it as command argument...?

**Comment by CendioOssman** [comment]
No, I'm afraid there is no way right now. `vncpasswd` should be possible to run on Windows, but you are right that we do not currently build it.

An argument is unfortunately very insecure, so it's not something we want to do. Environment variables are sometimes okay, but not in general. On Unix systems it's common to have invoke a new command for the password (e.g. `SSH_ASKPASS`). But I'm not sure what's common on Windows.

**Comment by dbolton** [comment]
Environment variables work on Windows too. 

Ideally passwords are saved in the Windows Credential Store, which ensures the password is encrypted at rest (but available to the current user). For example here's what Microsoft did for Git's version of askpass: https://microsoft.github.io/Git-Credential-Manager-for-Windows/Docs/Askpass.html

**Comment by CendioOssman** [comment]
Note that we now support specifying the password via the environment (#958), which is a more secure alternative to this issue.

**Comment by MDAR** [comment]
Sorry to bump an old thread, but this seemed the most appropriate

I installed TigerVNC on a number of machines without issue, however today on a fresh Debian 12 machine, TigerVNC Scraping installed, but the important `vncpasswd` command isn't installed

Package installed today 
```
 apt-get install tigervnc-common tigervnc-scraping-server
```

Has the `vncpasswd` been moved to a different package?

**Comment by CendioOssman** [comment]
I'm afraid that is something you'll need to discuss with Debian as that is their packaging, and not ours. It is also off-topic here, as this issue discusses a client feature.
